### DISCUSSION TITLE
onStart calls the function twice

### DISCUSSION THREAD

**Discussion by PlacidSerene** [ndresx/react-countdown#214] [Q&A] [answered]
`<Countdown
        date={Date.now() + 2000}
        onStart={()=>console.log('start')}
 />`

I have noticed that onStart actually calls my function twice. How do I prevent it?

**Comment by ndresx** [comment, marked as answer]
Hi @PlacidSerene, could you please check how often `onMount` triggers for you? Or could you maybe provide more context please?

**Reply by PlacidSerene** [reply]
Hi @ndresx, I've checked the onMount as well and it seems that it triggers twice. A little bit about the thing I'm trying to do: I'm making a countdown app and when the countdown starts, some toasts will be shown up (I'm using react-toastify for it).  Everything works great but the onStart triggers twice. I've attached the sandbox below.

https://codesandbox.io/s/reverent-bird-ndihqf?file=/src/App.js

**Reply by ndresx** [reply]
Hi! This seems to be caused by React 18's StrictMode (https://reactjs.org/docs/strict-mode.html). Removing it from the index.js file will only trigger the lifecycle methods like mounting once. https://codesandbox.io/s/sweet-joji-1kqxsg?file=/src/App.js

**Reply by PlacidSerene** [reply]
Thank you it works!
